### ISSUE TITLE
[journald]: Failed to determine boots: No data available

### ISSUE THREAD

**Issue by shibumi** [systemd/systemd#4088]
### Submission type
- [X] Bug report
- [ ] Request for enhancement (RFE)

_NOTE: Do not submit anything other than bug reports or RFEs via the issue tracker!_
### systemd version the issue has been seen with

> 231

_NOTE: Do not submit bug reports about anything but the two most recently released systemd versions upstream!_
### Used distribution

> Arch Linux
### In case of bug report: Expected behaviour you didn't see

> I should get a list of all boots
### In case of bug report: Unexpected behaviour you saw

> I get the message `Failed to determine boots: No data available` but I can go to the boots via `--boot` without a problem. 
### In case of bug report: Steps to reproduce the problem

> Sorry I don't know

**Comment by poettering** [comment]
Hmm, what's the precise command you issued?

If this reproducible, could you supply me with your /var/log/journal that allow me to check this locally? (Either append them here, if you are not concerned about privacy, or send them to me personally, if that's OK).

**Comment by shibumi** [comment]
Oh, Sorry I should include the command in my bug report:

```
#journalctl --list-boots 
Failed to determine boots: No data available
```

But I can get every boot with `journalctl --boot 1` or any other number. I just can't list them.

Are you sure you want my journal? it's 1G.

**Comment by poettering** [comment]
@shibumi any chance you can copy it into a temporary directory, and then remove journal files until you get the minimum set of files that reproduce the issue? use journalctl's --directory= switch on that directory to make it operate on that copy of the journal files...

**Comment by shibumi** [comment]
@poettering 
Ok, I have identified the journal file now. It's just 16mb. I will upload it somewhere and send you the link via query in IRC in freenode. This file definitly reproduce this error.

**Comment by poettering** [comment]
On Mon, 10.10.16 09:59, Christian Rebischke (notifications@github.com) wrote:

> @poettering 
> Ok, I have identified the journal file now. It's just 16mb. I will upload it somewhere and send you the link via query in IRC in freenode. This file definitly reproduce this error.

email much preferred: lennart at poettering dot net

Lennart

**Comment by shibumi** [comment]
@poettering 
You should have an email. If not it's possible that it's in your spam folder due to the 15-20mb journal file.

**Comment by poettering** [comment]
yeah, got the mail, thanks

**Comment by poettering** [comment]
Thanks for providing me with the file. This was very helpful! #4362 should make things work for you.

**Comment by shibumi** [comment]
Nice, can you summarize the reason the Bug?

**Comment by poettering** [comment]
this is the relevant commit:

https://github.com/systemd/systemd/pull/4362/commits/989793d341e730f452175fa18cf0f7ef4529d62c

It has some explanation in the commit msg and in the comment... I just noticed it's full of spelling errors and stuff, but I hope you still get the gist. Basically, it adds code to properly iterate through half-synced journal files (i.e. journal files which only got writen to disk half when the power was turned off)
